File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/BalloonAnnotation.swift
//
//  BalloonAnnotation.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class BalloonAnnotation:TKExamplesExampleViewController {
    
    let chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        let months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun"]
        let values = [95, 40, 55, 30, 76, 34]
        var array = [TKChartDataPoint]()
        for i in 0..<months.count {
            array.append(TKChartDataPoint(x: months[i], y: values[i]))
        }
        let series = TKChartLineSeries(items: array)
        
        // >> chart-point-shape-swift
        series.style.pointShape = TKPredefinedShape(type: TKShapeType.Circle, andSize: CGSizeMake(10, 10))
        // << chart-point-shape-swift
        
        // >> chart-point-pallete-swift
        let paletteItem = TKChartPaletteItem()
        paletteItem.fill = TKSolidFill(color: UIColor.redColor())
        let palette = TKChartPalette()
        palette.addPaletteItem(paletteItem)
        series.style.shapePalette = palette
        // << chart-point-pallete-swift
        
        chart.addSeries(series)
        
        // >> chart-balloon-annotation-swift
        let paragraphStyle = NSParagraphStyle.defaultParagraphStyle().mutableCopy() as! NSMutableParagraphStyle
        paragraphStyle.alignment = NSTextAlignment.Center
        
        let attributedText = NSMutableAttributedString(string: "Important milestone:\n $55000",
            attributes: [NSForegroundColorAttributeName:UIColor.whiteColor(), NSParagraphStyleAttributeName:paragraphStyle])
        attributedText.addAttribute(NSForegroundColorAttributeName, value: UIColor.yellowColor(), range: NSMakeRange(22, 6))
        
        var balloon = TKChartBalloonAnnotation(x: "Mar", y: 55, forSeries: series)
        balloon.attributedText = attributedText
        balloon.style.distanceFromPoint = 20
        balloon.style.arrowSize = CGSizeMake(10, 10)
        chart.addAnnotation(balloon)
        // << chart-balloon-annotation-swift
        
        balloon = TKChartBalloonAnnotation(text: "The lowest value:\n $30000", x: "Apr", y: 30, forSeries: series)
        balloon.style.verticalAlign = TKChartBalloonVerticalAlignment.Bottom
        chart.addAnnotation(balloon)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

File: Examples/Examples/Docs/Annotations/BalloonAnnotation.cs
using System;
using System.Collections.Generic;
using System.Drawing;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("BalloonAnnotation")]
	public class BalloonAnnotation: XamarinExampleViewController
	{	
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			TKChart chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			string[] months = new string[]{ "Jan", "Feb", "Mar", "Apr", "May", "Jun" };
			int[] values = new int[]{ 95, 40, 55, 30, 76, 34 };
			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();

			for (int i = 0; i < months.Length; i++) {
				list.Add (new TKChartDataPoint(new NSString(months[i]), new NSNumber(values[i])));
			}

			TKChartLineSeries series = new TKChartLineSeries (list.ToArray());

			// >> chart-point-shape-cs
			series.Style.PointShape = new TKPredefinedShape (TKShapeType.Circle, new SizeF (10, 10));
			// << chart-point-shape-cs

			// >> chart-point-pallete-cs
			var paletteItem = new TKChartPaletteItem ();
			paletteItem.Fill = new TKSolidFill (UIColor.Red);
			var palette = new TKChartPalette ();
			palette.AddPaletteItem (paletteItem);
			series.Style.ShapePalette = palette;
			// << chart-point-pallete-cs

			chart.AddSeries (series);
			// >> chart-balloon-annotation-cs
			NSMutableParagraphStyle paragraphStyle = (NSMutableParagraphStyle)new NSParagraphStyle ().MutableCopy();
			paragraphStyle.Alignment = UITextAlignment.Center;
			NSMutableDictionary attributes = new NSMutableDictionary ();
			attributes.Add (UIStringAttributeKey.ForegroundColor, UIColor.White);
			attributes.Add (UIStringAttributeKey.ParagraphStyle, paragraphStyle);
			NSMutableAttributedString attributedText = new NSMutableAttributedString ("Important milestone:\n $55000", attributes);

			attributedText.AddAttribute (UIStringAttributeKey.ForegroundColor, UIColor.Yellow, new NSRange (22, 6));

			TKChartBalloonAnnotation balloon = new TKChartBalloonAnnotation (new NSString("Mar"), new NSNumber(55), series);
			balloon.AttributedText = attributedText;
			balloon.Style.DistanceFromPoint = 20;
			balloon.Style.ArrowSize = new Size (10, 10);
			chart.AddAnnotation (balloon);
			// << chart-balloon-annotation-cs

			balloon = new TKChartBalloonAnnotation ("The lowest value:\n $30000", new NSString("Apr"), new NSNumber(30), series);
			balloon.Style.VerticalAlign = TKChartBalloonVerticalAlignment.Bottom;
			chart.AddAnnotation (balloon);
		}
	}
}

